agentops/instrumentation/common/logs.py

```diff
@@ -0,0 +1,86 @@
+import builtins
+import logging
+import os
+import atexit
+from datetime import datetime
+from typing import Any, TextIO
+from agents import Span
+
+_original_print = builtins.print
+
+LOGFILE_NAME = "agentops-tmp.log"
+
+## Instrument loggers and print function to log to a file
+def setup_print_logger() -> None:
+    """
+    ~Monkeypatches~ *Instruments the built-in print function and configures logging to also log to a file.
+    Preserves existing logging configuration and console output behavior.
+    """
+    log_file = os.path.join(os.getcwd(), LOGFILE_NAME)
+    
+    file_logger = logging.getLogger('agentops_file_logger')
+    file_logger.setLevel(logging.DEBUG)
+    
+    file_handler = logging.FileHandler(log_file)
```

Comment:
rather than creating a temp file could we just store the data to a variable?

we have to read the entire contents into memory for the upload anyway

Reply:
that's a very fair point. I figured in case the run crashes but we dump on exception anyway